src/webgpu/api/validation/resource_usages/textureUsageInRender.spec.ts

```diff
@@ -574,3 +577,64 @@ g.test('bindings_in_bundle')
       encoder.finish();
     }, !success);
   });
+
+g.test('unused_bindings_in_pipeline')
+  .params(params().combine(pbool('useBindGroup0')).combine(pbool('useBindGroup1')))
+  .fn(async t => {
+    const { useBindGroup0, useBindGroup1 } = t.params;
+    const view = t.createTexture({ usage: GPUTextureUsage.STORAGE }).createView();
+    const bindGroup0 = t.createBindGroup(0, view, 'readonly-storage-texture', 'rgba8unorm');
+    const bindGroup1 = t.createBindGroup(0, view, 'writeonly-storage-texture', 'rgba8unorm');
+
+    const wgslVertex = `
+      fn main() -> void {
+      }
```

Comment:
This requires a 'return' statement at the end.

Reply:
Done.